cudax/include/cuda/experimental/__utility/driver_api.cuh

```diff
@@ -0,0 +1,84 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDAX__UTILITY_DRIVER_API
+#define _CUDAX__UTILITY_DRIVER_API
+
+#include <cuda.h>
+
+#include <cuda/std/__exception/cuda_error.h>
+
+// Get the driver function by name using this macro
+#define CUDAX_GET_DRIVER_FUNCTION(function_name) \
+  reinterpret_cast<decltype(function_name)*>(get_driver_entry_point(#function_name))
+
+namespace cuda::experimental::detail::driver
```

Comment:
I am not a fan of the `__detail` namespace either because it is just a convention that users should not rely on it.

Sorry for the bad joke :) In the end it does not matter how we name our internal namespaces, users will pull out the guts of header-only libraries eventually anyway.

Reply:
Actually `__detail` is a reserved keyword, so any user that uses that is on their own if something goes wrong

Reply:
I know. But, I can imagine many users not knowing that. Or just still reaching into such a namespace, because they just need to add this one workaround (that eventually becomes production code and then haunts everybody forever). If it's not a compilation error (and I know some compilers have warnings for that), we cannot protect ourselves and we are just left with conventions.

Having said that, I don't mind whether the namespace is called `detail`, `__detail` or `internal`, etc. I just wanted to state that `__detail` buys us little over `detail`.